18.md

```diff
@@ -16,41 +16,66 @@ Payer MAY attach a `payerid` query parameter to LNURL-PAY callback with value se
 
 JSON array may contain one or more records of the following forms:
 
-1. `[ "text/plain", <free-form string> ]`
-2. `[ "application/pubkey", hex(<randomly generated secp256k1 pubkey>) ]`
-3. `[ "application/lnurl-auth", hex(<linkingKey>), hex(sign(utf8ToBytes(<k1>), <linkingPrivKey>)) ]`
-4. `[ "text/identifier", <internet identifier> ]`
-5. `[ "text/email", <email address> ]`
+1. `{ "tag": "name", "value": <free-form string> }`
+2. `{ "tag": "pubkey", "value": hex(<randomly generated secp256k1 pubkey>) }`
+3. `{ "tag": "auth", "value": hex(<linkingKey>), "sig": hex(sign(utf8ToBytes(<k1>), <linkingPrivKey>)) }` // following [LUD-04](04.md)
+4. `{ "tag": "identifier", "value": <internet identifier> }`
+5. `{ "tag": "email", "value": <email address> }`
+6. other fields can be negotiated.
```

Comment:
To make things more consistent with the `payerIDs` field in the request sent by the service, how about having the `payerid` parameter like this?

```
[
  "name": <free form string>,
  "pubkey": hex(<randomly generated secp256k1 pubkey>),
  "auth": {
    "value": hex(<linkingKey>),
    "sig": hex(sign(utf8ToBytes(<k1>), <linkingPrivKey>)) // following LUD-04
  },
  "email": <email address>,
  "identifier": <internet identifier>
]
```

And I also think it should be stated that this json must be then stringified (maybe remove spaces/blank chars) and url encoded. And that this **exact value** will be used later when checking the `SERVICE`'s invoice. It's not really obvious at first.

Reply:
@dpad85 It doesn't have to be stringified though? Definitely url encoded, but that's expected anyway.

Agree on changing the format.